engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/StartPlayingScreen.java

```diff
@@ -16,18 +16,93 @@
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.texture.Texture;
 import org.terasology.rendering.nui.CoreScreenLayer;
 import org.terasology.rendering.nui.WidgetUtil;
+import org.terasology.rendering.nui.widgets.UIImage;
+import org.terasology.rendering.world.WorldSetupWrapper;
+import org.terasology.world.internal.WorldInfo;
+import org.terasology.world.time.WorldTime;
 
 public class StartPlayingScreen extends CoreScreenLayer {
 
+    @In
+    private ModuleManager moduleManager;
+
+    @In
+    private Config config;
+
+    @In
+    private GameEngine gameEngine;
+
     public static final ResourceUrn ASSET_URI = new ResourceUrn("engine:startPlayingScreen");
+    private Texture texture;
+    private WorldSetupWrapper world;
+    private UniverseWrapper universeWrapper;
 
     @Override
     public void initialise() {
 
         WidgetUtil.trySubscribe(this, "close", button ->
                 triggerBackAnimation()
         );
+
+        WidgetUtil.trySubscribe(this, "play", button -> {
+            GameManifest gameManifest = new GameManifest();
+
+            gameManifest.setTitle(universeWrapper.getGameName());
+
+            DependencyResolver resolver = new DependencyResolver(moduleManager.getRegistry());
+            ResolutionResult result = resolver.resolve(config.getDefaultModSelection().listModules());
+            if (!result.isSuccess()) {
+                MessagePopup errorMessagePopup = getManager().pushScreen(MessagePopup.ASSET_URI, MessagePopup.class);
+                if (errorMessagePopup != null) {
+                    errorMessagePopup.setMessage("Invalid Module Selection", "Please review your module seleciton and try again");
+                }
+                return;
+            }
+            for (Module module : result.getModules()) {
+                gameManifest.addModule(module.getId(), module.getVersion());
+            }
+
+            SimpleUri uri = world.getWorldGeneratorInfo().getUri();
+            float timeOffset = 0.25f + 0.025f;
+            WorldInfo worldInfo = new WorldInfo(TerasologyConstants.MAIN_WORLD, world.getWorldGenerator().getWorldSeed(),
+                    (long) (WorldTime.DAY_LENGTH * timeOffset), uri);
+            gameManifest.addWorld(worldInfo);
+            gameEngine.changeState(new StateLoading(gameManifest, (universeWrapper.getLoadingAsServer()) ? NetworkMode.DEDICATED_SERVER : NetworkMode.NONE));
+        });
+    }
+
+    @Override
+    public void onOpened() {
+        UIImage previewImage = find("preview", UIImage.class);
+        previewImage.setImage(texture);
+    }
+
+    /**
+     * This method is called before the screen comes to the forefront to set the world
+     * in which the player is about to spawn.
+     * @param targetWorld The world in which the player is going to spawn.
+     * @param targetWorldTexture The world texture generated in {@link WorldPreGenerationScreen} to be displayed on this
+     *                           screen.
```

Comment:
You can probably just throw this one word back on the main line to keep it a one liner. I sometimes finding myself rewriting javadoc slightly to make more one liners for how nicely it lines up stuff :-)